### ISSUE TITLE
NDS Emulation - Stylus Controls

### ISSUE THREAD

**Issue by Lopezmadrid7** [knulli-cfw/distribution#59]
### Knulli build version

knulli-h700-rg35xx-plus-40-20240530.img.gz

### Your architecture

RG35XX Plus

### Issue description

First off absolutely amazing work with the CFW, fully have my support in saying best OS for the RG products. 

My issue consists of a known previous issue particularly with Drastic to run DS games. Initially I had issues with games not booting using the 'auto' function, this is resolved by selecting 'Drastic' as a core. My ultimate issue which I could not fix is the button mapping, particularly when needing to use the stylus controls to get past small 'touch' section such as SM64 NDS. I tried all the previous hotkeys, including holding R2 (and tried every other button for that matter), all of which did not work for me. My back buttons only worked to switch screen layouts. I have tried the following troubleshooting: 

- Flashing main SD card again with the last knulli version 
- Adding the ES NDS config file previously provided to the emu-station shared folder

In summary, I cannot get the touch screen/stylus buttons to work. 

### Detailed reproduction steps

Re-installed Knulli with Rufus (also detailed above)

### Details of any attempts to fix this yourself

See main description

### Details of any modifications you have made to Knulli.

Nil

**Comment by ropeholster** [comment]
Can confirm. I flashed to Knulli from Batocera v39 today and had this issue after having no issues with the DS stylus previously. It looks like all this is caused by is the fact that drastic.cfg has different input mappings in this build. I flashed back to v39 to compare, copied drastic.cfg from that build and overwrote Knulli's version of that file with the previous file, however it seems like the OS overwrites this file on boot each time. I wasn't able to find what setting / file causes that.

Hopefully this helps someone smarter than me figure out the fix here.

**Comment by ropeholster** [comment]
Can confirm that this is still an issue with the newest release. The stylus click now works, however it seems like the keys to move the stylus are unmapped at the moment.

**Comment by antiNT2** [comment]
I think it's Menu + Select to enable the stylus. In my opinion it should be R2

**Comment by Mikhailzrick** [comment]
In the 7/21 public build some left over files from the sans-joystick devices got left in for the devices with joysticks. So, it should not be an issue with the plus/sp/etc (hotkey+select switches dpad mode to virtual joystick). The dpad switch is on a system level it can’t be changed specifically for drastic.

for the devices with joysticks you can manually swap out the evmapy file from the GitHub or manually edit it, then and save an overlay via batocera-save-overlay. It’s in usr/share/evmapy/nds.drastic.keys

Other than that the issue has been resolved already.
